pkg/kds/remote/components.go

```diff
@@ -83,18 +88,24 @@ func providedFilter(clusterName string) reconcile.ResourceFilter {
 	}
 }
 
-func Callbacks(rt core_runtime.Runtime, syncer sync_store.ResourceSyncer, k8sStore bool, localZone string) *kds_client.Callbacks {
+func Callbacks(rt core_runtime.Runtime, syncer sync_store.ResourceSyncer, k8sStore bool, localZone string, kubeFactory resources_k8s.KubeFactory) *kds_client.Callbacks {
 	return &kds_client.Callbacks{
 		OnResourcesReceived: func(clusterID string, rs model.ResourceList) error {
-			if k8sStore &&
-				rs.GetItemType() != mesh.MeshType &&
-				rs.GetItemType() != system.SecretType &&
-				rs.GetItemType() != system.ConfigType {
-				util.AddSuffixToNames(rs.GetItems(), "default")
+			if k8sStore && rs.GetItemType() != system.ConfigType && rs.GetItemType() != system.SecretType {
+				// if type of Store is Kubernetes then we want to store upstream resources in dedicated Namespace.
+				// KubernetesStore parses Name and considers substring after the last dot as a Namespace's Name.
+				// System resources are not in the kubeFactory therefore we need explicit ifs for them
```

Comment:
Where is explicit "ifs"? I didn't manage to find it

Reply:
4 lines above
```
system.ConfigType && rs.GetItemType() != system.SecretType
```